### DISCUSSION TITLE
Dynamic import module is not defined error with Webpack5

### DISCUSSION THREAD

**Discussion by edew** [vercel/next.js#33607] [Help]
### Run `next info` (available from version 12.0.8 and up)

    /bin/sh: yarn: command not found
    /bin/sh: pnpm: command not found

    Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 20.6.0: Wed Nov 10 22:23:07 PST 2021; root:xnu-7195.141.14~1/RELEASE_X86_64
    Binaries:
      Node: 14.17.0
      npm: 6.14.13
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.0.8
      react: 17.0.2
      react-dom: 17.0.2

### What version of Next.js are you using?

12.0.8

### What version of Node.js are you using?

14.17.0

### What browser are you using?

Chrome 97.0.4692.99 (Official Build) (x86_64)

### What operating system are you using?

macOS

### How are you deploying your application?

N/A

### Describe the Bug

The development mode (`npm run dev`) build errors because `module is not defined` in a dynamically imported file.
 

### Expected Behavior

There should be no error and the dynamically imported component should render.

### To Reproduce

1. Clone [this repository](https://github.com/edew/nextjs-webpack-module-bug
)
1. Run `npm run dev`
1. Open http://localhost:3000
1. Enable "Pause on caught exceptions" or equivalent feature in your browser.

**Comment by balazsorban44** [comment]
There are a few issues here.

1. You are on an older `canary` version of Next.js. You should investigate using the newer `canary`/`latest` versions before concluding as issues have been fixed since then.
2. Your module contains JSX, which should be transpiled as it is not an ESM feature. I assume you were trying to create a simple reproduction, but I do want to mention if the module is in the same project as Next.js, you don't need the `package.json` at all.
3. Your `package.json` contains a non-standard `module` field to export `Foo`. Please refer to the Node.js documentation to understand how to do this correctly. I just gave a few pointers in https://github.com/vercel/next.js/issues/33599#issuecomment-1019995460 which are relevant.

**Comment by edew** [comment]
Seems to be fixed by https://github.com/vercel/next.js/pull/33637

> Fix issue with client-side JavaScript showing module is not defined

🎉
